packages/api/src/controllers/experiment/attestation.ts

```diff
@@ -50,15 +58,64 @@ app.post("/", validatePost("attestation"), async (req, res) => {
   return res.status(201).json(attestationMetadata);
 });
 
-function verifySigner(message, signature): boolean {
-  const verifiedSigner = ethers.verifyTypedData(
-    DOMAIN,
-    TYPES,
-    message,
-    signature
-  );
+async function verifySignature(
+  message: any,
+  signature: string,
+  signatureType?: string
+): Promise<string> {
+  const verifiedEIP712 =
+    (signatureType === "eip712" || !signatureType) &&
+    verifyEIP712Signature(message, signature);
+  const verifiedFlow =
+    (signatureType === "flow" || !signatureType) &&
+    (await verifyFlowSignature(message, signature));
+
+  return verifiedEIP712 ? "eip712" : verifiedFlow ? "flow" : null;
+}
+
+function verifyEIP712Signature(message: any, signature: string): boolean {
+  try {
+    const verifiedSigner = ethers.verifyTypedData(
+      DOMAIN,
+      TYPES,
+      message,
+      signature
+    );
+
+    return verifiedSigner === message.signer;
+  } catch (e) {
+    return false;
+  }
+}
 
-  return verifiedSigner === message.signer;
+async function verifyFlowSignature(
+  message: any,
+  signature: string
+): Promise<boolean> {
+  try {
+    // These parameters are required to use the fcl library, even though we don't use on-chain verification
+    await fcl.config({
+      "flow.network": "testnet",
+      "accessNode.api": "https://access-testnet.onflow.org",
+    });
```

Comment:
Do we/Can we call this config function multiple times, before every query? Might wanna consider moving it to app initialization e.g. `app-router.ts` or sth like that

Reply:
Moved in 429881b1073e849eccde9c7359b4015167bfbaa2